pyteal/ast/abi/reference_type.py

```diff
@@ -25,6 +27,9 @@ class Account(Uint):
     def __init__(self) -> None:
         super().__init__(AccountTypeSpec())
 
+    def deref(self) -> Expr:
+        return Txn.accounts[self.stored_value.load()]
```

Comment:
Probably the following are non-issues and the onus is on the PyTeal developer to understand the value of implicit index 0, but it's worth clarifying. And if that's the case, maybe there's still a way to reduce foot shooting:

* Is this logic going to have an off-by-one problem because index-0 implicitly refers to the self account/app (see [this arc-4 section](https://github.com/algorandfoundation/ARCs/blob/main/ARCs/arc-0004.md#reference-types)) ? 
* For assets the logic would be different?